ambry-protocol/src/main/java/com.github.ambry.protocol/GetResponse.java

```diff
@@ -131,6 +133,26 @@ public long writeTo(WritableByteChannel channel) throws IOException {
     return written;
   }
 
+  @Override
+  public void writeTo(AsyncWritableChannel channel, Callback callback) throws IOException {
+    if (bufferToSend == null) {
+      bufferToSend = ByteBuffer.allocate(
+          (int) super.sizeInBytes() + (Partition_Response_Info_List_Size + partitionResponseInfoSize));
+      writeHeader();
+      if (partitionResponseInfoList != null) {
+        bufferToSend.putInt(partitionResponseInfoList.size());
+        for (PartitionResponseInfo partitionResponseInfo : partitionResponseInfoList) {
+          partitionResponseInfo.writeTo(bufferToSend);
+        }
+      }
+      bufferToSend.flip();
+    }
+    channel.write(bufferToSend, callback);
+    if (toSend != null) {
+      toSend.writeTo(channel, callback);
+    }
```

Comment:
same here, callback should only be invoked once.

Reply:
Will make another PR to address this issue. For this PR, I will set callback to null in child send.